plugins/src/core/index.ts

```diff
@@ -0,0 +1,369 @@
+import * as cordova from "cordova";
+import {
+  PushNotificationHandler,
+  InAppDeepLinkHandler,
+  InAppInboxUpdatedHandler,
+  DeepLinkHandler,
+} from "./handlers";
+
+export interface OptimoveConfig{
+  pushReceivedHandler?: PushNotificationHandler | null;
+  pushOpenedHandler?: PushNotificationHandler | null;
+  inAppDeepLinkHandler?: InAppDeepLinkHandler | null;
+  inAppInboxUpdatedHandler?: InAppInboxUpdatedHandler | null;
+  deepLinkHandler?: DeepLinkHandler | null;
+}
+
+let currentConfig: OptimoveConfig = {
+  pushReceivedHandler: null,
+  pushOpenedHandler: null,
+  inAppDeepLinkHandler: null,
+  inAppInboxUpdatedHandler: null,
+  deepLinkHandler: null,
+};
+document.addEventListener("deviceready", init, false);
+document.addEventListener("resume", resume, false);
+document.addEventListener("pause", pause, false);
+
+function init() {
+  setContext();
+}
+
+function resume() {
+  setContext();
+}
+
+function pause() {
+  clearContext();
+}
+
+function setContext() {
+  setHandlersCallBackContext().then(
+    (successMessage) => {
+      console.log(successMessage);
+    },
+    (errorMessage) => {
+      console.error(errorMessage);
+    }
+  );
+}
+function clearContext() {
+  return new Promise((resolve, reject) => {
+    cordova.exec(resolve, reject, "OptimoveSDKPlugin", "clearContext", []);
+  });
+}
+
+function setHandlersCallBackContext() {
+  return new Promise((resolve, reject) => {
+    cordova.exec(
+      nativeMessageHandler,
+      reject,
+      "OptimoveSDKPlugin",
+      "setHandlersCallBackContext",
+      []
+    );
+  });
+}
+function checkIfPendingDDLExists() {
+  return new Promise((resolve, reject) => {
+    cordova.exec(
+      nativeMessageHandler,
+      reject,
+      "OptimoveSDKPlugin",
+      "checkIfPendingDDLExists",
+      []
+    );
+  });
+}
+
+function checkIfPendingPushExists() {
+  return new Promise((resolve, reject) => {
+    cordova.exec(
+      nativeMessageHandler,
+      reject,
+      "OptimoveSDKPlugin",
+      "checkIfPendingPushExists",
+      []
+    );
+  });
+}
+
+function nativeMessageHandler(message) {
+  if (!message || typeof message === "string") {
+    return;
+  }
+
+  const handlerName = `${message.type}Handler`;
+  if (
+    handlerName === "inAppInboxUpdatedHandler" &&
+    typeof currentConfig[handlerName] === "function"
+  ) {
+    currentConfig[handlerName]?.();
+
+    return;
+  }
+
+  if (typeof currentConfig[handlerName] === "function") {
```

Comment:
can be replaced with `currentConfig[handlerName] !== null`. And same for all typeof